astropy/wcs/src/wcslib_prjprm_wrap.c

```diff
@@ -0,0 +1,1025 @@
+    if (copy == NULL) return NULL;
+    return (PyObject*)copy;
+}
+
+
+static PyObject* PyPrjprm_deepcopy(PyPrjprm* self)
+{
+    PyPrjprm* copy = PyPrjprm_new(&PyPrjprmType, NULL, NULL);
+    if (copy == NULL) return NULL;
+
+    memcpy(copy->x, self->x, sizeof(struct prjprm));
+    copy->x->err = NULL;
+    return (PyObject*)copy;
+}
+
+
+static PyObject* PyPrjprm___str__(PyPrjprm* self)
+{
+    wcsprintf_set(NULL);
+    if (wcslib_prj_to_python_exc(prjprt(self->x))) {
+        return NULL;
+    }
+    return PyUnicode_FromString(wcsprintf_buf());
+}
+
+
+static int PyPrjprm_cset(PyPrjprm* self)
+{
+    if (wcslib_prj_to_python_exc(prjset(self->x))) {
+        return -1;
+    }
+    return 0;
+}
+
+
+static PyObject* PyPrjprm_set(PyPrjprm* self)
+{
+    if (is_readonly(self) || PyPrjprm_cset(self)) return NULL;
+    Py_RETURN_NONE;
+}
+
+
+static PyObject* _prj_eval(PyPrjprm* self, int (*prjfn)(PRJX2S_ARGS),
+                           PyObject* x1_in, PyObject* x2_in)
+{
+    Py_ssize_t i, ndim;
+    npy_intp *x1_dims, *x2_dims;
+    Py_ssize_t     nelem      = 1;
+    PyArrayObject* x1         = NULL;
+    PyArrayObject* x2         = NULL;
+    PyArrayObject* prj_x1     = NULL;
+    PyArrayObject* prj_x2     = NULL;
+    PyArrayObject* stat       = NULL;
+    PyObject*      result     = NULL;
+    int            status     = -1;
+
+    // TODO: This assumes the same shape for the input arrays.
+    //       Instead, we should broadcast.
+
+    x1 = (PyArrayObject *) PyArray_ContiguousFromObject(x1_in, NPY_DOUBLE, 1, NPY_MAXDIMS);
+    if (x1 == NULL) {
+      goto exit;
+    }
+    x2 = (PyArrayObject *) PyArray_ContiguousFromObject(x2_in, NPY_DOUBLE, 1, NPY_MAXDIMS);
+    if (x2 == NULL) {
+      goto exit;
+    }
+
+    ndim = PyArray_NDIM(x1);
+
+    if (ndim != PyArray_NDIM(x2)) {
+        PyErr_SetString(PyExc_ValueError, "Input array dimensions do not match.");
+        goto exit;
+    }
+
+    x1_dims = PyArray_DIMS(x1);
+    x2_dims = PyArray_DIMS(x2);
+    for (i = 0; i < ndim; i++) {
+        if (x1_dims[i] != x2_dims[i]) {
+            PyErr_SetString(PyExc_ValueError, "Input array dimensions do not match.");
+            goto exit;
+        }
+        nelem *= x1_dims[i];
+    }
+
+    prj_x1 = (PyArrayObject*)PyArray_SimpleNew(ndim, x1_dims, NPY_DOUBLE);
+    if (prj_x1 == NULL) {
+      goto exit;
+    }
+
+    prj_x2 = (PyArrayObject*)PyArray_SimpleNew(ndim, x1_dims, NPY_DOUBLE);
+    if (prj_x2 == NULL) {
+      goto exit;
+    }
+
+    stat = (PyArrayObject*)PyArray_SimpleNew(ndim, x1_dims, NPY_INT);
+    if (stat == NULL) {
+      goto exit;
+    }
+
+    Py_BEGIN_ALLOW_THREADS
+    status = prjfn(
+        self->x,
+        nelem, 0, 1, 1,
+        (double*)PyArray_DATA(x1),
+        (double*)PyArray_DATA(x2),
+        (double*)PyArray_DATA(prj_x1),
+        (double*)PyArray_DATA(prj_x2),
+        (int*)PyArray_DATA(stat));
+    Py_END_ALLOW_THREADS
+
+    switch (status) {
+    case 3:
+    case 4:
```

Comment:
The docstrings for prjx2s and prjs2x state that InvalidCoordinateError is raised, but this looks like it's returning arrays of NAN with no exception?

Reply:
This is what the rest of `astropy.wcs` is doing in this situation. So, I am just being consistent here: when coordinate conversion fails, coordinate transformation returns `np.nan`.